go-update/src/main/java/org/reactome/release/goupdate/DuplicateCleaner.java

```diff
@@ -0,0 +1,182 @@
+package org.reactome.release.goupdate;
+
+import java.io.FileInputStream;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.gk.model.GKInstance;
+import org.gk.model.ReactomeJavaConstants;
+import org.gk.persistence.MySQLAdaptor;
+import org.gk.schema.GKSchemaAttribute;
+import org.reactome.release.common.ReleaseStep;
+
+/**
+ * Stand-alone class with main method that can be used to clean up duplicate GO accessions.
+ * @author sshorser
+ *
+ */
+public class DuplicateCleaner extends ReleaseStep
+{
+	private static final Logger logger = LogManager.getLogger();
+
+	public static void main(String[] args)
+	{
+		String pathToResources = "src/main/resources/go-update.properties";
+		if (args.length > 0)
+		{
+			pathToResources = args[0];
+		}
+		try
+		{
+			Properties props = new Properties();
+			props.load(new FileInputStream(pathToResources));
+			DuplicateCleaner step = new DuplicateCleaner();
+			step.executeStep(props);
+		}
+		catch (Exception e)
+		{
+			e.printStackTrace();
+		}
+
+	}
+
+	@Override
+	public void executeStep(Properties props) throws Exception
+	{
+		MySQLAdaptor adaptor = DuplicateCleaner.getMySQLAdaptorFromProperties(props);
+		int totalDuplicatesCount = 0;
+		int instancesWithSignificantReferrers = 0;
+		
+		DuplicateReporter dupeReporter = new DuplicateReporter(adaptor);
+		Set<Long> dbIDsToDelete = new HashSet<Long>();
+
+		Map<String, Integer> duplicates = dupeReporter.getDuplicateAccessions();
+		
+		// loop through all duplicated accessions.
+		for (String accession : duplicates.keySet())
+		{
+			totalDuplicatesCount = duplicates.keySet().size();
```

Comment:
Should this be `+=` instead of `=` since the count is used after the for loop is finished?

Reply:
Or could the `totalDuplicatesCount` variable be removed and the logger statement using it just call `duplicates.keySet().size()` directly?